packages/doenetml-worker-rust/lib-doenetml-core/src/core/general_prop/component_ref_prop.rs

```diff
@@ -0,0 +1,95 @@
+use std::rc::Rc;
+
+use crate::{
+    components::prelude::*,
+    props::ContentFilter,
+    state::types::{component_refs::ComponentRef, content_refs::ContentRef},
+};
+
+/// A prop that references a single component
+#[derive(Debug)]
+pub struct ComponentRefProp {
+    /// The data query that indicates how the dependencies of this prop will be created.
+    data_query: DataQuery,
+
+    elements_to_select: Option<ElementsToSelect>,
```

Comment:
`element_to_select`

Or maybe if we are using `component` rather than `element` now, then `component_to_select`.